### PR TITLE
add edge deployment provider resources

### PR DESCRIPTION
**Pull request by shawnps**
Upgrades `go-sigsci` to `v0.1.10` and adds edge deployment resources.

### CODE REVIEW

**Comment by jonnangle** [comment]
Thanks for this - we've been working on doing this orchestration with a local-exec provisioner, so it's great to see this in the provider!

A couple of things I've noticed whilst testing:

- destroying a `sigsci_edge_deployment_service` errors out, because [the `DELETE` API call returns HTTP 200](https://docs.fastly.com/signalsciences/api/#_corps__corpName__sites__siteName__edgeDeployment__fastlySID__delete) but [go-sigsci expects to see a 204](https://github.com/signalsciences/go-sigsci/blob/main/api.go#L127-L130)
- the ID for the `sigsci_edge_deployment_service` is the site shortname - is this right? My understanding is that there's a many to one relationship between Fastly services and Signal Sciences sites, so it feels like the ID should be linked to the Fastly service ID?
- what's your expectation for activating the new version of the Fastly service that is created by the `sigsci_edge_deployment_service` resource? Should we plan on activating that ourselves manually?

File: docs/resources/edge_deployment_service.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+---
+
+# sigsci_edge_deployment_service (Resource)
+
+
+
+## Example Usage
+
+```terraform
+resource "sigsci_site" "my-site" {
```
Comment (jonnangle) [inline docs/resources/edge_deployment_service.md:16]:
```suggestion
resource "sigsci_edge_deployment_service" "my-service" {
```

File: examples/resources/sigsci_edge_deployment_service/resource.tf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+resource "sigsci_site" "my-site" {
```
Comment (jonnangle) [inline examples/resources/sigsci_edge_deployment_service/resource.tf:1]:
```suggestion
resource "sigsci_edge_deployment_service" "my-service" {
```

**Comment by shawnps** [comment]
@jonnangle thank you very much for taking a look at this, I appreciate the feedback! Also thank you for the example fixes.

> destroying a sigsci_edge_deployment_service errors out, because [the DELETE API call returns HTTP 200](https://docs.fastly.com/signalsciences/api/#_corps__corpName__sites__siteName__edgeDeployment__fastlySID__delete) but [go-sigsci expects to see a 204](https://github.com/signalsciences/go-sigsci/blob/main/api.go#L127-L130)

Thanks -- I've made a PR [here](https://github.com/signalsciences/go-sigsci/pull/41) to allow both status 200 and 204 for DELETE calls.

> the ID for the sigsci_edge_deployment_service is the site shortname - is this right? My understanding is that there's a many to one relationship between Fastly services and Signal Sciences sites, so it feels like the ID should be linked to the Fastly service ID?

Good question, I had thought it was a one-to-one relationship, I'll do some more investigation on this one.

> what's your expectation for activating the new version of the Fastly service that is created by the sigsci_edge_deployment_service resource? Should we plan on activating that ourselves manually?

I'll confirm with the team but I think we'll need to update the [`fastly_service_vcl`](https://github.com/fastly/terraform-provider-fastly/blob/main/docs/resources/service_vcl.md) schema in the Fastly TF provider to allow this.

**Review by jonnangle — APPROVED** [review]
Looks good to me, thanks!

**Comment by jonnangle** [comment]
In terms of activating the Fastly version - thanks for the info! We'll keep an eye on developments in the provider, and will probably use a local-exec provisioner to activate the new version and keep us ticking over in the meantime.

File: provider/config.go [hunk 3]
```diff
@@ -24,11 +25,28 @@ func (c *Config) Client() (interface{}, error) {
...
+		err    error
+	)
+
 	if c.APIToken != "" {
-		return sigsci.NewTokenClient(c.Email, c.APIToken), nil
+		client = sigsci.NewTokenClient(c.Email, c.APIToken)
 	} else if c.Password != "" {
-		return sigsci.NewClient(c.Email, c.Password)
+		client, err = sigsci.NewClient(c.Email, c.Password)
+
```
Comment (freeformz) [inline provider/config.go:41]: nit: drop the extra line

File: provider/resource_edge_deployment.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+				Type:        schema.TypeString,
+				Description: "Site short name",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func createOrUpdateEdgeDeployment(d *schema.ResourceData, m interface{}) error {
+	pm := m.(providerMetadata)
```
Comment (freeformz) [inline provider/resource_edge_deployment.go:25]: what happens when m is not a providerMetadata? Is that possible?

File: provider/resource_edge_deployment.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+	return sc.CreateOrUpdateEdgeDeployment(pm.Corp, d.Get("site_short_name").(string))
+}
+
+func readEdgeDeployment(d *schema.ResourceData, m interface{}) error {
+	return nil
+}
+
+func deleteEdgeDeployment(d *schema.ResourceData, m interface{}) error {
+	pm := m.(providerMetadata)
```
Comment (freeformz) [inline provider/resource_edge_deployment.go:38]: what happens when m is not a providerMetadata? Is that possible?

File: provider/resource_edge_deployment.go [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+				Description: "Site short name",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func createOrUpdateEdgeDeployment(d *schema.ResourceData, m interface{}) error {
+	pm := m.(providerMetadata)
+	sc := pm.Client
```
Comment (freeformz) [inline provider/resource_edge_deployment.go:26]: Why copy the client?

File: provider/resource_edge_deployment.go [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	return sc.CreateOrUpdateEdgeDeployment(pm.Corp, d.Get("site_short_name").(string))
+}
+
+func readEdgeDeployment(d *schema.ResourceData, m interface{}) error {
+	return nil
+}
+
+func deleteEdgeDeployment(d *schema.ResourceData, m interface{}) error {
+	pm := m.(providerMetadata)
+	sc := pm.Client
```
Comment (freeformz) [inline provider/resource_edge_deployment.go:39]: Why copy the client?

**Review by freeformz — APPROVED** [review]

File: provider/resource_edge_deployment.go (hunk 4 above)
Comment (shawnps) [reply provider/resource_edge_deployment.go:25]: Good question, to be honest I copied this logic from existing resources. It looks like the `interface{}` returned by `provider.ConfigureFunc` is passed into the resources:

https://github.com/signalsciences/terraform-provider-sigsci/blob/main/provider/provider.go#L69
https://github.com/signalsciences/terraform-provider-sigsci/blob/main/provider/provider.go#L95

https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk@v1.14.0/helper/schema#ConfigureFunc

File: provider/resource_edge_deployment.go (hunk 5 above)
Comment (shawnps) [reply provider/resource_edge_deployment.go:38]: See https://github.com/signalsciences/terraform-provider-sigsci/pull/112#discussion_r1107840500

File: provider/resource_edge_deployment.go (hunk 6 above)
Comment (shawnps) [reply provider/resource_edge_deployment.go:26]: This is another thing I got from existing resources. I'll change it to `pm.Client.CreateOrUpdateEdgeDeployment` instead.

**Pull request merged**